pbench_runner/pbench-uperf.wrapper.py

```diff
@@ -0,0 +1,240 @@
+#   v0.0.2 - 03/12/2021 - Write ret.stdout to a file.
+#   v0.0.3 - 03/12/2021 - Use re to check parameter 'config' correct or not.
+# Reversions:
+#   v0.0.1 - 03/01/2021 - Built the script.
+#   v0.1.0 - 03/03/2021 - Supported 'group' different commands of click.
+#   v0.2.0 - 03/04/2021 - Supported test suites to run test.
+#   v0.3.0 - 03/06/2021 - Fixed int parameters from YAML.
+#   v0.4.0 - 03/07/2021 - Removed 'group' of click.
+#   v0.5.0 - 03/09/2021 - Handled parameters values None or not.
+#   v0.5.1 - 03/12/2021 - Enhanced script comments.
+#   v0.6.0 - 03/12/2021 - Fixed full test suites parameters values when YAML file was changed.
+
+
+import os
+import click
+import yaml
+import subprocess
+
+
+# YAML file name.
+file_name = "pbench-uperf.job"
+
+
+# Read a pbench-uperf.job YAML which owns full test cases by default.
+def get_yaml(file_name):
+    # To find a pbench-uperf.job file located the same folder with the script.
+    job_path = os.path.abspath(".")
+    job_file = os.path.join(job_path, file_name)
+    if os.path.isfile(job_file) and os.path.exists(job_file):
+        print("INFO: Job file - {} exists.".format(file_name))
+    else:
+        print("ERROR: Exit as file has been gone.")
+        exit 1
+
+    # Open target YAML file.
+    with open(job_file, "r", encoding = "utf-8") as jf:
+        job_content = yaml.load(jf.read(), Loader=yaml.Loader)
+        # print("DEBUG: job_content:\n{}".format(job_content))
+
+    # Below keys are the parameters from the YAML file to configure pbench-uperf test.
+    config = job_content["config"]
+    print(config)
+    
+    # Type of protocols is a list.
+    protocols = job_content["protocols"]
+    protocols = ",".join(protocols)
+    print(protocols)
+
+    # Type of test_types is a list.
+    test_types = job_content["test-types"]
+    test_types = ",".join(test_types)
+    print(test_types)
+
+    runtime = job_content["runtime"]
+    print(runtime)
+
+    # Convert int elements of list to str, as pbench-uperf is a shell script.
+    message_sizes = job_content["message_sizes"]
+    message_sizes = [str(x) for x in message_sizes]
+    # Type of message_sizes is a list.
+    message_sizes = ",".join(message_sizes)
+    print(message_sizes)
+
+    # Convert int elements of list to str, as pbench-uperf is a shell script.
+    instances = job_content["instances"]
+    instances = [str(x) for x in instances]
+    # Type of instances is a list.
+    instances = ",".join(instances)
+    print(instances)
+
+    nr_samples = job_content["nr_samples"]
+    print(nr_samples)
+
+    max_failures = job_content["max_failures"]
+    print(max_failures)
+
+    maxstddevpct = job_content["maxstddevpct"]
+    print(maxstddevpct)
+
+    return protocols, config, test_types, runtime, message_sizes, instances, nr_samples, max_failures, maxstddevpct
+
+
+# Define three test suites for pbench-uperf-quick-tests.
+def test_suites(test_suite_name):
```

Comment:
determine_suites_params()

Remove the yaml file, just leave the unspecified args as "none", let pbench-uperf determine them. (keep the same logic).